### PR TITLE
bugfixes: bloodcrawl, losebreath, wintercoat, and more

### LINKED ISSUE #9686: Emag not working on accessible airlocks properly
Clicking an airlock you have access to with the emag causes it to open normally rather than become emagged. If you click again, the airlock closes, and then it sparks and re-opens emagged. If you double click an airlock you have access to, you can sometimes get it to instantly pop open without the animation, so you can immediately run through it without having to wait for it to open.

### LINKED ISSUE #11799: Ghosts can possess drones before round start.
This is primarily bad because of how much round start code assumes things.

I wouldn't be shocked if you could break something that way.

### LINKED ISSUE #11869: Dead shadowlings receive "The light burns you!" messages even while dead
Hatched shadowlings, to be clear.

How to reproduce:

- Be shadowling
- Die
- Be dead in an area with light
- Be spammed

I'm not sure if this is actually trying to decrement health, all I saw were the messages.

### LINKED ISSUE #11924: Luchadore mask vs winter coat hood (sprites)
http://i.imgur.com/W20AJhB.png

Something is on the wrong layer or has the wrong cover flags. Shouldn't be displaying like this.

### LINKED ISSUE #11942: Admin healing/reviving a morph makes it invisible
If its disguised as something. This seems to persist between form changes sometimes as well.

### LINKED ISSUE #11962: Heart attacks are permanent.
Once heart_attack has been set to 1, the human will take oxy damage forever. Setting heart_attack back to 0 won't stop it.

Death will stop the damage, but once the human is revived, damage will continue.

### LINKED ISSUE #11995: Coming out of blood leaves you unable to hear things being said.
After coming out of blood after eating a slaughter demon heart, you are left deaf and unable to hear anything being said by anyone. It does not even show anything you say, along with no message that usually indicates you can't hear anything. It's as if nothing ever is being said despite people saying things. I was able to repeat it a second time after it occurred to me. No clue if this also applies to actual slaughter demons.

### LINKED ISSUE #12125: Cyborgs have a strip menu
Probably already been reported because it's been around forever (couldn't find it in the issue tracker though), but cyborgs have a strip menu that just lists left and right hands that you can't interact with because they don't have any.

### PR DESCRIPTION
**Pull request by phil235**
- Fixes shadowling receiving burn messages even when dead. Fixes #11869. I've also fixed the same bug for other species and made certain spec_life() procs not run when the mob is dead. I also made slime species a subtype of jelly species since they had a lot of code in common thus reducing duplicate code.
- Fixes odd behavior with emagging airlock and windoor animation (the doors are now "busy" during the emagging). Fixes #9686
- Fixes being able to bloodcrawl in oil decal.
- Fixes being deaf after bloodcrawling. Fixes #11995 
- Fixes wintercoat hood appearing below the mask. Fixes #11924
- Fixes being able to attack another mob before the game has started (and other things which used a gamestart check that no longer works). 
- Ghosts can no longer become drones before the game has started. Fixes #11799.
- Removed "bhunger" and "ajourn" mob vars, they are unused.
- Fixes admin-revived morph being invisible. Fixes #11942
- Fixes heart attack stacking losebreath very high. Also now losebreath is reset to 0 when the mob dies. Fixes #11962 
- All mobs no longer have a default strip menu showing two hands. Fixes #12125

### CODE REVIEW

**Comment by WJohn** [comment]
Praise be.

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -265,6 +266,9 @@
 	if (src.operating)
 		return
 
+	if(istype(I, /obj/item/weapon/card/emag))
+		return
```
Comment (phil235) [inline code/game/machinery/doors/windowdoor.dm]: not a great fix but I'm too lazy to rewrite the whole proc or more.

**Comment by RemieRichards** [comment]
@phil235 how did you fix #11924 ? I dont see anything about layers in your changes.

**Comment by phil235** [comment]
@RemieRichards I gave winter hoods actual sprites and removed the hood from the suit sprite of the winter coat. The issue was that it was the hood sprite on the mob was actually just the suit sprite, which has a layer below masks.

**Comment by RemieRichards** [comment]
Aha that does make sense. Good work.

**Comment by Dorsisdwarf** [comment]
>Ghosts can no longer become drones before the game has started. Fixes #11799.

Fun removal

File: code/modules/mob/living/carbon/human/life.dm [hunk 2]
```diff
@@ -335,7 +335,8 @@
 	if(!heart_attack)
 		return
 	else
-		losebreath += 5
+		if(losebreath < 3)
```
Comment (Iamgoofball) [inline code/modules/mob/living/carbon/human/life.dm]: ehhh, this should probably be 10 at most

just to make sure I can't completely nullify it with oxyloss meds

**Comment by newfren** [comment]
Yeah fairly upset that you can't become a drone before roundstart, considering basically all that time is spent actually getting to the station anyway.

**Comment by duncathan** [comment]
@newfren sorry but having players before roundstart is really buggy

**Pull request merged**